src/microbe_stage/editor/MicrobeEditorActionData.cs

```diff
@@ -0,0 +1,345 @@
+        if (moveActionData.NewLocation == OldLocation)
+        {
+            return new MoveActionData(Organelle, moveActionData.OldLocation, NewLocation, moveActionData.OldRotation,
+                NewRotation);
+        }
+
+        return new MoveActionData(moveActionData.Organelle, OldLocation, moveActionData.NewLocation, OldRotation,
+            moveActionData.NewRotation);
+    }
+}
+
+[JSONAlwaysDynamicType]
+public class MembraneActionData : MicrobeEditorActionData
+{
+    public MembraneType OldMembrane;
+    public MembraneType NewMembrane;
+
+    public MembraneActionData(MembraneType oldMembrane, MembraneType newMembrane)
+    {
+        OldMembrane = oldMembrane;
+        NewMembrane = newMembrane;
+    }
+
+    public override MicrobeActionInterferenceMode GetInterferenceModeWith(MicrobeEditorActionData other)
+    {
+        if (other is MembraneActionData membraneActionData)
+        {
+            if (membraneActionData.NewMembrane == OldMembrane && NewMembrane == membraneActionData.OldMembrane)
+                return MicrobeActionInterferenceMode.CancelsOut;
+
+            if (membraneActionData.NewMembrane == OldMembrane || NewMembrane == membraneActionData.OldMembrane)
+                return MicrobeActionInterferenceMode.Combinable;
+        }
+
+        return MicrobeActionInterferenceMode.NoInterference;
+    }
+
+    public override int CalculateCost()
+    {
+        return NewMembrane.EditorCost;
+    }
+
+    protected override MicrobeEditorActionData CombineGuaranteed(MicrobeEditorActionData other)
+    {
+        var membraneActionData = (MembraneActionData)other;
+        if (OldMembrane == membraneActionData.NewMembrane)
+            return new MembraneActionData(membraneActionData.OldMembrane, NewMembrane);
+
+        return new MembraneActionData(membraneActionData.NewMembrane, OldMembrane);
+    }
+}
+
+[JSONAlwaysDynamicType]
+public class RigidityChangeActionData : MicrobeEditorActionData
+{
+    public float NewRigidity;
+    public float PreviousRigidity;
+
+    public RigidityChangeActionData(float newRigidity, float previousRigidity)
+    {
+        NewRigidity = newRigidity;
+        PreviousRigidity = previousRigidity;
+    }
+
+    public override MicrobeActionInterferenceMode GetInterferenceModeWith(MicrobeEditorActionData other)
+    {
+        if (other is RigidityChangeActionData rigidityChangeActionData)
+        {
+            if (Math.Abs(NewRigidity - rigidityChangeActionData.PreviousRigidity) < MathUtils.EPSILON &&
+                Math.Abs(rigidityChangeActionData.NewRigidity - PreviousRigidity) < MathUtils.EPSILON)
+                return MicrobeActionInterferenceMode.CancelsOut;
+
+            if (Math.Abs(NewRigidity - rigidityChangeActionData.PreviousRigidity) < MathUtils.EPSILON ||
+                Math.Abs(rigidityChangeActionData.NewRigidity - PreviousRigidity) < MathUtils.EPSILON)
+                return MicrobeActionInterferenceMode.Combinable;
```

Comment:
Can we safely use if/else here? Or just have it always be combinable, even if to a zero change?

Reply:
I'm sorry, I don't fully understand. I've added comments, maybe those are clarifying this code a bit?